### PR TITLE
SLVS-1314 SLCore 10.4 update, API changes & related fixes

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**

### CODE REVIEW

File: src/EmbeddedSonarAnalyzer.props [hunk 1]
```diff
@@ -9,6 +9,6 @@
     <EmbeddedSonarJSAnalyzerVersion>10.14.0.26080</EmbeddedSonarJSAnalyzerVersion>
     <EmbeddedSonarSecretsJarVersion>2.13.0.3515</EmbeddedSonarSecretsJarVersion>
     <!-- SLOOP: Binaries for SonarLint Out Of Process -->
-    <EmbeddedSloopVersion>10.3.2.78498</EmbeddedSloopVersion>
+    <EmbeddedSloopVersion>10.4.0.78808</EmbeddedSloopVersion>
```
Comment (vnaskos-sonar) [inline src/EmbeddedSonarAnalyzer.props:12]: As discussed update to latest released version

File: src/SLCore.IntegrationTests/RuleConfigurationAnalysisTests.cs [hunk 2]
```diff
@@ -31,6 +31,14 @@ public class RuleConfigurationAnalysisTests : FileAnalysisTestsBase
     private const string CtorParamRuleId = "javascript:S107";
     private const int ActualCtorParams = 4;
     private const string CtorParamName = "maximumFunctionParameters";
+    
+    protected override Guid AnalysisId { get; set; }
```
Comment (vnaskos-sonar) [inline src/SLCore.IntegrationTests/RuleConfigurationAnalysisTests.cs:35]: Make it beautiful 😄

File: src/SLCore.IntegrationTests/SimpleAnalysisTests.cs [hunk 3]
```diff
@@ -25,6 +25,14 @@ namespace SonarLint.VisualStudio.SLCore.IntegrationTests;
 [TestClass]
 public class SimpleAnalysisTests : FileAnalysisTestsBase
 {
+    protected override Guid AnalysisId { get; set; }
```
Comment (vnaskos-sonar) [inline src/SLCore.IntegrationTests/SimpleAnalysisTests.cs:28]: same as above https://github.com/SonarSource/sonarlint-visualstudio/pull/5571/files#r1698455858

File: src/SLCore/Listener/Analysis/IAnalysisConfigurationProviderListener.cs [hunk 4]
```diff
@@ -18,11 +18,18 @@
...
+public interface IAnalysisConfigurationProviderListener : ISLCoreListener
 {
     Task<GetBaseDirResponse> GetBaseDirAsync(GetBaseDirParams parameters);
+
+    Task<GetInferredAnalysisPropertiesResponse> GetInferredAnalysisPropertiesAsync(GetInferredAnalysisPropertiesParams parameters);
 }
+
+public record GetInferredAnalysisPropertiesResponse(Dictionary<string, string> properties);
+
+public record GetInferredAnalysisPropertiesParams(string configurationScopeId, List<FileUri> filePathsToAnalyze);
```
Comment (vnaskos-sonar) [inline src/SLCore/Listener/Analysis/IAnalysisConfigurationProviderListener.cs]: Move to separate files

File: src/SLCore.IntegrationTests/FileAnalysisTestsBase.cs [hunk 5]
```diff
@@ -133,7 +136,9 @@ private static async Task RunSlCoreFileAnalysis(SLCoreTestRunner slCoreTestRunne
 
     protected static string GetFullPath(string fileToAnalyzeRelativePath)
     {
-        return Path.GetFullPath(fileToAnalyzeRelativePath);
+        var currentDomainBaseDirectory = new DirectoryInfo(AppDomain.CurrentDomain.BaseDirectory);
+        var integrationTestBaseDirectory = currentDomainBaseDirectory.Parent.Parent.Parent.FullName;
```
Comment (vnaskos-sonar) [inline src/SLCore.IntegrationTests/FileAnalysisTestsBase.cs:140]: This feels a bit unsafe. Can be replaced maybe by
```
var solutionPath= Path.GetDirectoryName(Path.GetDirectoryName(TestContext.TestDir));
return Path.Combine(solutionPath, @"src\SLCore.IntegrationTests", fileToAnalyzeRelativePath);
```

**Review by vnaskos-sonar — APPROVED** [review]
All good, just one comment/concern.

File: src/SLCore.IntegrationTests/FileAnalysisTestsBase.cs (hunk 5 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.IntegrationTests/FileAnalysisTestsBase.cs:140]: This is hardcoding the integration test project name. I will stick with the AppDomain approach as it seems more stable

**Pull request merged**